src/system/freertos_plus_tcp/network.c

```diff
@@ -0,0 +1,106 @@
+#include <stdlib.h>
+
+#include "zenoh-pico/system/platform.h"
+#include "zenoh-pico/utils/pointers.h"
+#include "zenoh-pico/utils/result.h"
+
+// FreeRTOS includes
+#include "FreeRTOS.h"
+#include "FreeRTOS_IP.h"
+
+#if Z_LINK_UDP_UNICAST == 1 || Z_LINK_UDP_MULTICAST == 1
+/*------------------ UDP sockets ------------------*/
+int8_t _z_create_endpoint_udp(_z_sys_net_endpoint_t *ep, const char *s_address, const char *s_port) {
+    int8_t ret = _Z_RES_OK;
+
+    if (FreeRTOS_getaddrinfo(s_address, NULL, NULL, &ep->_iptcp) < 0) {
+        ret = _Z_ERR_GENERIC;
+    }
+
+    ep->_iptcp->ai_addr->sin_family = FREERTOS_AF_INET4;
```

Comment:
isn't this value set by `getaddrinfo`?

Reply:
Unfortunately not, `getaddrinfo` only sets `ai_family` but does not write it to `ai_addr->sin_family`.  I changed it so it at least uses the value returned by getaddrinfo.